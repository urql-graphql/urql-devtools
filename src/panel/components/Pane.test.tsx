jest.mock("../hooks", () => ({
  useOrientationWatcher: jest.fn(),
}));

import React from "react";
import { shallow } from "enzyme";
import { useOrientationWatcher } from "../hooks";
import { Pane } from "./Pane";

const useOrientation = (useOrientationWatcher as jest.Mocked<any>);
const addEventListener = jest.spyOn(window, "addEventListener");

beforeEach(jest.clearAllMocks);

describe("on mount", () => {
  describe("on portrait orientation", () => {
    beforeEach(() => {
      useOrientation.mockReturnValue({ isPortrait: true, isLandscape: false });
    });

    it("matches snapshot", () => {
      const wrapper = shallow(<Pane />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("on landscape orientation", () => {
    beforeEach(() => {
      useOrientation.mockReturnValue({
        isPortrait: false,
        isLandscape: true,
      });
    });

    it("matches snapshot", () => {
      const wrapper = shallow(<Pane />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});

describe("on mouse down", () => {
  beforeEach(() => {
    const wrapper = shallow(<Pane />);
    wrapper
      .find("DraggingEdge")
      .simulate("mouseDown", { button: 0, preventDefault: jest.fn() });
  });
  it("listens for mouse up events", () => {
    expect(addEventListener).toBeCalledWith("mouseup", expect.any(Function));
  });

  it("listens for mouse move events", () => {
    expect(addEventListener).toBeCalledWith("mousemove", expect.any(Function));
  });
});
