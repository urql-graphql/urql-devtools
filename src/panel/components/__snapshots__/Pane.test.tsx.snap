// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`on mount on landscape orientation matches snapshot 1`] = `
<PaneContainer
  data-portrait="false"
  style={
    {
      "height": "auto",
      "minWidth": 400,
      "width": 400,
    }
  }
>
  <DraggingEdge
    aria-grabbed={false}
    aria-orientation="vertical"
    data-portrait="false"
    onMouseDown={[Function]}
    role="seperator"
  />
</PaneContainer>
`;

exports[`on mount on portrait orientation matches snapshot 1`] = `
<PaneContainer
  data-portrait="true"
  style={
    {
      "height": 400,
      "minHeight": 400,
      "width": "auto",
    }
  }
>
  <DraggingEdge
    aria-grabbed={false}
    aria-orientation="horizontal"
    data-portrait="true"
    onMouseDown={[Function]}
    role="seperator"
  />
</PaneContainer>
`;
